src/Dfc.CourseDirectory.WebV2/Features/NewApprenticeshipProvider/ProviderDetailInfoPanel.cs

```diff
@@ -67,21 +57,31 @@ public async Task<ViewModel> Handle(Query request, CancellationToken cancellatio
                 .OrderByDescending(c => c.LastUpdated)
                 .SingleOrDefault(c => c.ContactType == "P");
 
-            var contactName = contact?.ContactPersonalDetails?.PersonGivenName != null && contact?.ContactPersonalDetails?.PersonFamilyName != null ?
+            if (contact == null)
+            {
+                return new ViewModel()
+                {
+                    ProviderId = request.ProviderId,
+                    GotContact = false
+                };
+            }
+
+            var contactName = contact.ContactPersonalDetails?.PersonGivenName != null && contact.ContactPersonalDetails?.PersonFamilyName != null ?
                 string.Join(" ", contact.ContactPersonalDetails.PersonGivenName) + " " +
                     contact.ContactPersonalDetails.PersonFamilyName :
                 null;
 
             return new ViewModel()
```

Comment:
Consider `ViewModel.FromProvider(Provider provider)...` method to remove some of this logic from the controller. (suggestion, not request)

Reply:
The current V2 approach is having dumb VM classes (POCOs) then having the mapping logic in these handler types.